src/state/defeat.rs

```diff
@@ -9,5 +17,46 @@ pub fn update(app: &mut App, state: &mut State) {
         state.stage = Stage::Playing;
         state.elapsed_milisec = 0;
         state.board.reset();
+        state.explosions.clear();
+    }
+
+    if state.explosions.is_empty() {
```

Comment:
This whole code should probably just be run once on state transition (in the `playing` module)